nebula-logger/core/main/logger-engine/lwc/logger/loggerService.js

```diff
@@ -7,234 +7,130 @@ import { newLogEntry } from './logEntryBuilder';
-                  resolve(LoggerService.settings);
-              });
+    async _loadSettingsFromServer() {
+        try {
+            const settings = await getSettings();
+            this.#settings = Object.freeze({
+                ...settings,
+                supportedLoggingLevels: Object.freeze(settings.supportedLoggingLevels),
+                userLoggingLevel: Object.freeze(settings.userLoggingLevel)
+            });
+        } catch (error) {
+            /* eslint-disable-next-line no-console */
+            console.error(error);
+            throw error;
+        }
     }
 
     _meetsUserLoggingLevel(logEntryLoggingLevel) {
-        if (LoggerService.settings && LoggerService.settings.supportedLoggingLevels && LoggerService.settings.userLoggingLevel) {
-            const currentIsEnabled =
-                LoggerService.settings.isEnabled === true &&
-                LoggerService.settings.userLoggingLevel.ordinal <= LoggerService.settings?.supportedLoggingLevels[logEntryLoggingLevel];
-            return currentIsEnabled ? LOADING_ENUM.enabled : LOADING_ENUM.disabled;
-        }
-        return LOADING_ENUM.loading;
+        return this.#settings.isEnabled === true && this.#settings.userLoggingLevel.ordinal <= this.#settings?.supportedLoggingLevels[logEntryLoggingLevel];
     }
 
     _newEntry(loggingLevel, message) {
         // Builder is returned immediately but console log will be determined after loading settings from server
-        const logEntryBuilder = newLogEntry(loggingLevel, this._loadSettingsFromServer);
+        const logEntryBuilder = newLogEntry(loggingLevel, this.#settings?.isConsoleLoggingEnabled);
         logEntryBuilder.setMessage(message);
         if (this.#scenario) {
             logEntryBuilder.scenario = this.#scenario;
         }
-        const loggingPromise = this._loadSettingsFromServer().then(() => {
-            const isEnabledEnum = this._meetsUserLoggingLevel(loggingLevel);
-            if (isEnabledEnum === LOADING_ENUM.enabled || isEnabledEnum === LOADING_ENUM.loading) {
-                const componentLogEntry = logEntryBuilder.getComponentLogEntry();
-                componentLogEntry.loadingEnum = isEnabledEnum;
-                this.#componentLogEntries.push(componentLogEntry);
-            }
-            if (this.#isSavingLog) {
-                this.#isSavingLog = false;
-                this.saveLog();
-            }
-        });
-        this.#loggingPromises.push(loggingPromise);
+        if (this._meetsUserLoggingLevel(loggingLevel)) {
+            this.#componentLogEntries.push(logEntryBuilder.getComponentLogEntry());
+        }
 
         return logEntryBuilder;
     }
 };
 
-/**
- * @return {LoggerService} a LoggerService instance
- */
-const createLoggerService = function () {
-    return new LoggerService();
+const createLoggerService = async function () {
+    const service = new LoggerService();
+    await service._loadSettingsFromServer();
```

Comment:
@jamessimone I was thinking about the function `_loadSettingsFromServer()` - right now, if there are multiple instances of `logger` on a page (e.g., multiple LWCs are on the same page, and each LWC uses `logger`), then each call to `createLoggerService()` will try to call the Apex method `getSettings()`. But the result of `getSettings()` will be the same for every call, since it returns settings for the current user.

The `getSettings()` Apex method is marked as `cacheable=true`, so maybe it's fine as-is. But the current version of the LWC stores [the settings object in a static variable](https://github.com/jongpie/NebulaLogger/blob/v4.12.9/nebula-logger/core/main/logger-engine/lwc/logger/loggerService.js#L18), which seems more effective. I tried reintroducing this into the current branch, but using a `static` variable breaks multiple jest tests, because the `static` variable isn't cleared between jest tests.

Any thoughts? Do you think we should leave it as-is? Or do you see a way to cache the result of the Apex method (without breaking jest tests)?

Reply:
I'd let it be handled by the controller method being annotated with `cacheable=true` for now. I think that should suffice!

Reply:
Sounds good, I'll leave it as-is. Thanks!